test/test_jit.py

```diff
@@ -11188,6 +11188,21 @@ def return_tuple(x):
             return a, x
         self.checkScript(return_tuple, (torch.rand(4),))
 
+    def test_add_tuple_slice(self):
+        def foo(input: Tuple[torch.Tensor, Optional[torch.Tensor], Optional[torch.Tensor]]):
+            changed_input = input[0] + 1
+            return (changed_input,) + input[1:]
+        inp: Tuple[torch.Tensor, Optional[torch.Tensor], Optional[torch.Tensor]] = (torch.rand(4), None, None)
+        self.checkScript(foo, (inp,))
+
+    def test_add_tuple_slice_non_optional(self):
+        def foo(input: Tuple[torch.Tensor, torch.Tensor, torch.Tensor]):
+            changed_input = input[0] + 1
+            return (changed_input,) + input[1:]
```

Comment:
Can you add an example where the return type of the tuple add is used inside of a ScriptFunction ? `AnyTupleType` refers to `tuple` which is not type that you can do anything with inside of script function, it's kind of just an abstract supertype of all tuples. 

I think you're going to need to type the return value more accurately than `AnyTupleType` to be useable in TorchScript

Reply:
+1

Current test cases don't actually verify the output type is computed correctly. The case suggested by Elias would cover it.

Or, even better you can use FileCheck to verify the type is correct in generated IR

Reply:
I replaced them with return statements like value[3] or torch.sum(value[3]).